src/components/Panel/Actions/Actions.tsx

```diff
@@ -0,0 +1,152 @@
+import { ConnectButton, DecimalInput } from "@/components";
+import { connectWallet, settled, smallMobileAndUnder } from "@/constants";
+import { usePageContext, usePrimaryPanelAction } from "@/hooks";
+import type { OracleQueryUI } from "@/types";
+import Pencil from "public/assets/icons/pencil.svg";
+import Settled from "public/assets/icons/settled.svg";
+import type { CSSProperties } from "react";
+import { useState } from "react";
+import styled from "styled-components";
+import { useAccount, useNetwork } from "wagmi";
+import { SectionTitle, SectionTitleWrapper, Text } from "../style";
+import { ActionDetails } from "./ActionDetails";
+import { Errors } from "./Errors";
+import { Message } from "./Message";
+import { PrimaryActionButton } from "./PrimaryActionButton";
+
+interface Props {
+  query: OracleQueryUI;
+}
+export function Actions({ query }: Props) {
+  const [proposePriceInput, setProposePriceInput] = useState("");
+  const [inputError, setInputError] = useState("");
+  const { chainId, oracleType, valueText, actionType } = query;
+
+  const primaryAction = usePrimaryPanelAction({
+    query,
+    proposePriceInput,
+  });
+
+  const { address } = useAccount();
+  const { chain: connectedChain } = useNetwork();
+  const { page } = usePageContext();
+
+  const pageIsPropose = page === "propose";
```

Comment:
I added these booleans to make it easier to reason about these states. logic is the same